localization/util/pose_initializer/src/pose_initializer_core.cpp

```diff
@@ -146,18 +152,17 @@ void PoseInitializer::callbackGNSSPoseCov(
   add_height_pose_msg_ptr->pose.covariance[4 * 6 + 4] = 0.01;
   add_height_pose_msg_ptr->pose.covariance[5 * 6 + 5] = 3.14;
 
-  geometry_msgs::PoseWithCovarianceStamped::Ptr aligned_pose_msg_ptr(
-    new geometry_msgs::PoseWithCovarianceStamped);
-  const bool succeeded_align = callAlignService(*add_height_pose_msg_ptr, aligned_pose_msg_ptr);
+  auto aligned_pose_msg_ptr = std::make_shared<geometry_msgs::msg::PoseWithCovarianceStamped>();
```

Comment:
There is a possible improvement here that I didn't implement to keep it focused on the ROS2 port – this could live on the stack instead, and then be used as an out parameter by passing in a raw pointer.